softlayer/vm/fakes/fake_vm.go

```diff
@@ -82,6 +82,10 @@ func (vm *FakeVM) ReloadOS(stemcell bslcstemcell.Stemcell) error {
 	return vm.ReloadOSErr
 }
 
+func (vm *FakeVM) ReloadOSForBaremetal(string, string) error {
+	return vm.ReloadOSErr
```

Comment:
you should also save the parameters, so something like:
```
func (vm *FakeVM) ReloadOSForBaremetal(stemcell string, netbootImage string) error {
  vm.ReloadOSStemcell = stemcell
  vm.ReloadOSNetBootImage = netbootImage

  return vm.ReloadOSErr
}
```

Which is good practice as it allows you to verify params in.